apps/full-site-editing/full-site-editing-plugin/lib/feature-flags/feature-flags.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+class A8C_Full_Site_Editing_Feature_Flags {
+	private $flags = array();
+
+	public function __construct() {
+		$this->set_flags();
```

Comment:
Should this be a singleton? So we don't set the flags multiple times during the request lifetime.

Reply:
I'd like that. It would probably be beneficial to base it on the plugin architecture that #33250 introduces, and make it available to all FSE plugins

Reply:
> Should this be a singleton?

Yup, it makes sense.

> base it on the plugin architecture that #33250 introduces, and make it available to all FSE plugins

Yup, it makes sense x2. 🙂
I worked on this while y'all were doing #33250, but I had already in mind to wait and rebase on that after it got merged.

Reply:
Just noting that this was converted to singleton in previous commits.